apps/electron/layers/main/src/index.ts

```diff
@@ -15,7 +27,20 @@ if (!isSingleInstance) {
   process.exit(0);
 }
 
-app.on('second-instance', restoreOrCreateWindow);
+app.on('second-instance', (event, argv) => {
+  restoreOrCreateWindow();
+});
+
+app.on('open-url', async (_, url) => {
+  const mainWindow = BrowserWindow.getAllWindows().find(w => !w.isDestroyed());
+  const urlObj = parse(url, true);
+  console.log('window-open', url);
+  if (!mainWindow || !urlObj.query.code) return;
+  const token = (await exchangeToken(urlObj.query.code as string)) as {
+    id_token: string;
+  };
+  mainWindow.webContents.send('send-token', token.id_token);
```

Comment:
please give a more descriptive name to `send-token`, like `auth:callback-firebase-token`